server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -784,6 +817,103 @@ public List<Field> createFields(String name, Number value, boolean indexed, bool
+            @Override
+            public BigInteger parse(Object value, boolean coerce) {
+                return objectToUnsignedLong(value, coerce);
+            }
+
+            @Override
+            public Number parsePoint(byte[] value) {
+                return BigIntegerPoint.decodeDimension(value, 0);
+            }
+
+            @Override
+            public BigInteger parse(XContentParser parser, boolean coerce) throws IOException {
+                return parser.bigIntegerValue(coerce);
+            }
+
+            @Override
+            public Query termQuery(String field, Object value) {
+                if (hasDecimalPart(value)) {
+                    return Queries.newMatchNoDocsQuery("Value [" + value + "] has a decimal part");
+                }
+                BigInteger v = parse(value, true);
+                return BigIntegerPoint.newExactQuery(field, v);
+            }
+
+            @Override
+            public Query termsQuery(String field, List<Object> values) {
+                BigInteger[] v = new BigInteger[values.size()];
+                int upTo = 0;
+
+                for (int i = 0; i < values.size(); i++) {
+                    Object value = values.get(i);
+                    if (!hasDecimalPart(value)) {
+                        v[upTo++] = parse(value, true);
+                    }
+                }
+
+                if (upTo == 0) {
+                    return Queries.newMatchNoDocsQuery("All values have a decimal part");
+                }
+                if (upTo != v.length) {
+                    v = Arrays.copyOf(v, upTo);
+                }
+
+                return BigIntegerPoint.newSetQuery(field, v);
+            }
+
+            @Override
+            public Query rangeQuery(
+                String field,
+                Object lowerTerm,
+                Object upperTerm,
+                boolean includeLower,
+                boolean includeUpper,
+                boolean hasDocValues,
+                QueryShardContext context
+            ) {
+                return unsignedLongRangeQuery(lowerTerm, upperTerm, includeLower, includeUpper, (l, u) -> {
+                    Query query = BigIntegerPoint.newRangeQuery(field, l, u);
+                    if (hasDocValues) {
+                        Query dvQuery = SortedUnsignedLongDocValuesRangeQuery.newSlowRangeQuery(field, l, u);
+                        query = new IndexOrDocValuesQuery(query, dvQuery);
+                    }
+                    return query;
+                });
+            }
+
+            @Override
+            public List<Field> createFields(String name, Number value, boolean indexed, boolean docValued, boolean stored) {
+                List<Field> fields = new ArrayList<>();
+                final BigInteger v = Numbers.toUnsignedLongExact(value);
+
+                if (indexed) {
+                    fields.add(new BigIntegerPoint(name, v));
```

Comment:
Exactly, the `SortType` is the show stopper here sadly :( I kept this particular feature (and index sort as well), all because of the `SortType` limitation